Math/src/main/java/org/tribuo/math/neighbor/NeighborsQuery.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tribuo.math.neighbor;
+
+import com.oracle.labs.mlrg.olcut.util.Pair;
+import org.tribuo.math.la.SGDVector;
+
+import java.util.List;
+
+/**
+ * An interface for nearest neighbor queries.
+ */
+public interface NeighborsQuery {
```

Comment:
Should this extend `com.oracle.labs.mlrg.olcut.config.Configurable`?

Reply:
A new interface `NeighboursQueryFactory` and concrete implementation `NeighboursBruteForceFactory` has been added to address this discussion.